kdbg/dbgmainwnd.cpp

```diff
@@ -884,10 +884,13 @@ void DebuggerMainWnd::slotFileGlobalSettings()
 	if (m_asmGlobalFlavor != oldGlobalFlavor) {
 	    if (m_debugger != 0) {
 		/*
-		 * When a valid debugger is set, run this. If not,
-		 * this function will run again at debugProgram
+		 * When a valid debugger is set, run this. If not, this
+		 * function will run again at debugProgram. Also, don't
+		 * update the disassembly view (if any) since global flavor
+		 * shouldn't affect immediately the current flavor.
```

Comment:
Can you explain why the global flavor should not affect the displayed disassembly? What if the user has chosen "use global setting" in *This Program* and then modifies the global setting? In that case, I would expect that the display should be updated immediately.

Reply:
The global setting is supposed to be "visible" the next time KDbg will restart. But I hadn't considered the case you mentioned. I will look into this.

Reply:
About overwritting the m_flavor. Why not? I mean it can be useful to have "stored" the flavor string that `m_d->executeCmd(DCsetdisassflavor...` used. 

Also, about the "default" options I will use this string in both settings.

Reply:
> About overwritting the m_flavor. Why not? I mean it can be useful to have "stored" the flavor string that `m_d->executeCmd(DCsetdisassflavor...` used.

Because it will never be necessary to have the value stored somewhere. It can always be computed in the same way that `submitDisassemblyFlavor()` computed it. If the need arises, just factor out that determination into a new function `getEffectiveDisassemblyFlavor()`.